### PR TITLE
refactor: update verifier

### PR DESCRIPTION
**Pull request by Two-Hearts**
This PR updates the verifier to use `NewVerifierWithOptions` with `httputil.NewClient` as `OCSPHTTPClient`.

### CODE REVIEW

**Comment by JeyJeyGao** [comment]
LGTM, but pending on dependencies.

File: go.mod [hunk 1]
```diff
@@ -32,3 +32,7 @@ require (
 	golang.org/x/sync v0.6.0 // indirect
 	golang.org/x/sys v0.22.0 // indirect
 )
+
+replace github.com/notaryproject/notation-go => github.com/Two-Hearts/notation-go v0.0.0-20240801055637-02ac12ec085f
+
+replace github.com/notaryproject/notation-core-go => github.com/Two-Hearts/notation-core-go v0.0.0-20240801005628-17703be86fe1
```
Comment (shizhMSFT) [inline go.mod:38]: This comment is a reminder to remove the temp replacement

File: cmd/notation/verify.go [hunk 2]
```diff
@@ -219,3 +227,29 @@ func printMetadataIfPresent(outcome *notation.VerificationOutcome) {
...
+	x509TrustStore := truststore.NewX509TrustStore(dir.ConfigFS())
+	revocationCodeSigningValidator, err := revocation.NewWithOptions(revocation.Options{
+		OCSPHTTPClient:   httputil.NewClient(ctx, &http.Client{Timeout: 2 * time.Second}),
+		CertChainPurpose: x509.ExtKeyUsageCodeSigning,
+	})
+	if err != nil {
+		return nil, err
+	}
+	revocationTimestampingValidator, err := revocation.NewWithOptions(revocation.Options{
+		OCSPHTTPClient:   httputil.NewClient(ctx, &http.Client{Timeout: 2 * time.Second}),
```
Comment (shizhMSFT) [inline cmd/notation/verify.go:245]: Should the http client be reused?

Comment (Two-Hearts) [reply cmd/notation/verify.go]: Updated.

**Review by shizhMSFT — COMMENTED** [review]
LGTM except the dependency update

**Comment by rgnote** [comment]
LGTM. Will approve once mod files are updated.

File: go.mod (hunk 1 above)
Comment (Two-Hearts) [reply go.mod]: Removed.

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Review by rgnote — APPROVED** [review]
LGTM

**Pull request merged**